module/database/UserDatabase.py

```diff
@@ -65,7 +66,7 @@ def changePassword(db, user, oldpw, newpw):
         pw = r[2][5:]
         h = sha1(salt + oldpw)
         if h.hexdigest() == pw:
-            salt = reduce(lambda x, y: x + y, [str(random.randint(0, 9)) for i in range(0, 5)])
+            salt = functools.reduce(lambda x, y: x + y, [str(random.randint(0, 9)) for i in range(0, 5)])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=4846282)